app/frontend/wallet/shelley/helpers/poolCertificateUtils.ts

```diff
@@ -10,68 +34,64 @@ const enum PoolParamsByteLengths {
   METADATA_HASH = 32,
 }
 
-export type PoolOwnerParams = {
-  stakingPath?: any //BIP32Path
+type TxStakepoolOwner = {
   stakingKeyHashHex?: string
 }
 
-export type SingleHostIPRelay = {
-  portNumber?: number
-  ipv4?: string
-  ipv6?: string
-}
+export type TxStakepoolRelay = SingleHostIPRelay | SingleHostNameRelay | MultiHostNameRelay
 
-export type SingleHostNameRelay = {
-  portNumber?: number
-  dnsName: string
+type SingleHostIPRelay = {
+  type: TxRelayType.SINGLE_HOST_IP
+  params: {
+    portNumber?: number
+    ipv4?: string
+    ipv6?: string
+  }
 }
 
-export type MultiHostNameRelay = {
-  dnsName: string
+type SingleHostNameRelay = {
+  type: TxRelayType.SINGLE_HOST_NAME
+  params: {
+    portNumber?: number
+    dnsName: string
+  }
 }
 
-export type RelayParams = {
-  type: number // single host ip = 0, single hostname = 1, multi host name = 2
-  params: SingleHostIPRelay | SingleHostNameRelay | MultiHostNameRelay
+type MultiHostNameRelay = {
+  type: TxRelayType.MULTI_HOST_NAME
+  params: {
+    dnsName: string
+  }
 }
 
-export type PoolMetadataParams = {
+type TxStakepoolMetadata = {
   metadataUrl: string
   metadataHashHex: string
 }
 
-export type Margin = {
+type TxStakepoolMargin = {
   numeratorStr: string
   denominatorStr: string
 }
 
-export type PoolParams = {
-  poolKeyHashHex: string
-  vrfKeyHashHex: string
-  pledgeStr: string
-  costStr: string
-  margin: Margin
-  rewardAccountHex: string
-  poolOwners: Array<PoolOwnerParams>
-  relays: Array<RelayParams>
-  metadata: PoolMetadataParams
+// TODO: fix when migrating to BigInt
+export const checkedNumber = (number: any, variableName: string): number => {
```

Comment:
can the "number" at the input really be anything? It scares me a bit that for example Number(null) is zero, so this function would map a null value to zero without any warning

Reply:
Good point, I had no idea that Number(null) is 0.. that's pretty strange, anyway, fixed.